pkg/kubectl/cmd/certificates.go

```diff
@@ -0,0 +1,166 @@
+	}
+
+	cmd.AddCommand(NewCmdCertificateApprove(f, out))
+	cmd.AddCommand(NewCmdCertificateDeny(f, out))
+
+	return cmd
+}
+
+type CertificateOptions struct {
+	Recursive bool
+	Filenames []string
+}
+
+func NewCmdCertificateApprove(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := CertificateOptions{}
+	cmd := &cobra.Command{
+		Use:   "approve SUBCOMMAND",
+		Short: "approve a CSR.",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(RunCertificateApprove(f, out, cmd, args, &options))
+		},
+	}
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, "filenames of certificatesigningrequest objects to approve")
+	cmdutil.AddRecursiveFlag(cmd, &options.Recursive)
+	cmdutil.AddOutputFlagsForMutation(cmd)
+
+	return cmd
+}
+
+func RunCertificateApprove(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *CertificateOptions) error {
+	return modifyCertificateCondition(f, out, cmd, args, options, func(csr *certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, string) {
+		var alreadyApproved bool
+		for _, c := range csr.Status.Conditions {
+			if c.Type == certificates.CertificateApproved {
+				alreadyApproved = true
+			}
+
+		}
+		if alreadyApproved {
+			return csr, "approved"
+		}
+		csr.Status.Conditions = append(csr.Status.Conditions, certificates.CertificateSigningRequestCondition{
+			Type:           certificates.CertificateApproved,
+			Reason:         "KubectlApprove",
+			Message:        "This csr was approved by kubectl alpha approve",
+			LastUpdateTime: unversioned.Now(),
+		})
+		return csr, "approved"
+	})
+}
+
+func NewCmdCertificateDeny(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := CertificateOptions{}
+	cmd := &cobra.Command{
+		Use:   "deny SUBCOMMAND",
+		Short: "deny a CSR.",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(RunCertificateDeny(f, out, cmd, args, &options))
+		},
+	}
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, "filenames of certificatesigningrequest objects to deny")
+	cmdutil.AddRecursiveFlag(cmd, &options.Recursive)
+	cmdutil.AddOutputFlagsForMutation(cmd)
+
+	return cmd
+}
+
+func RunCertificateDeny(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *CertificateOptions) error {
+	return modifyCertificateCondition(f, out, cmd, args, options, func(csr *certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, string) {
+		var alreadyDenied bool
+		for _, c := range csr.Status.Conditions {
+			if c.Type == certificates.CertificateDenied {
+				alreadyDenied = true
+			}
+
+		}
+		if alreadyDenied {
+			return csr, "denied"
+		}
+		csr.Status.Conditions = append(csr.Status.Conditions, certificates.CertificateSigningRequestCondition{
+			Type:           certificates.CertificateDenied,
+			Reason:         "KubectlDeny",
+			Message:        "This csr was approved by kubectl alpha certificate deny",
```

Comment:
Saying the CLI command that denied it is a bit tacky.  I suspect we should support "--message" with a default that is "Denied by administrator (%s)" and substitute the full path from cmd.CommandPath() rather than hardcoding.